### ISSUE TITLE
Normalización min y max - Waldir Gregory Quispe Charaja

### ISSUE THREAD

**Issue by Waldir1995** [frdtorres/mainf24-2-BIDA#6]
from flask import Flask, jsonify
import pandas as pd
from sklearn.preprocessing import MinMaxScaler

app = Flask(__name__)

df = pd.read_csv('phone search.csv')
# Normalizar las columnas numéricas
scaler = MinMaxScaler()

# Seleccionamos solo las columnas numéricas que queremos normalizar
columnas_numericas = df.select_dtypes(include=['float64', 'int64']).columns

# Aplicamos la normalización a las columnas numéricas
df[columnas_numericas] = scaler.fit_transform(df[columnas_numericas])

print("Datos normalizados:")
print(df.head())

# Definir una ruta básica
@app.route('/')
def home():
    return "API de búsqueda de teléfonos con datos normalizados"

# Ruta para obtener todos los datos (normalizados)
@app.route('/api/phones', methods=['GET'])
def get_phones():
    data = df.to_dict(orient='records')  # Convertimos el DataFrame a diccionario
    return jsonify(data)  # Devolvemos los datos en formato JSON

# Ruta para obtener un teléfono específico por ID (normalizado)
@app.route('/api/phones/<int:id>', methods=['GET'])
def get_phone(id):
    phone = df[df['id'] == id].to_dict(orient='records')  # Filtramos el teléfono por ID
    if phone:
        return jsonify(phone[0])  # Devolvemos el teléfono encontrado en formato JSON
    return jsonify({'error': 'Teléfono no encontrado'}), 404  # Si no se encuentra, devolver un error 404

if __name__ == '__main__':
    app.run(debug=True)
